registry/backend.go

```diff
@@ -2,10 +2,13 @@ package registry
 
 type Backend interface {
 	// Register registers fabio as a service in the registry.
-	Register() error
+	Register(serviceName string) error
```

Comment:
I think a better approach would be to just extend the register function since you need to keep the state of what is registered in the backend anyway. 

```go
type Backend interface {
    Register(name ...string) error
    DeregisterAll() error
    ...
}
```